src/components/Expander/Expander.tsx

```diff
@@ -108,32 +108,62 @@ const IfTitleTag = ({
   </Fragment>
 );
 
-export class ExpanderItem extends Component<ExpanderProps> {
+class ExpanderItem extends Component<ExpanderProps> {
   state: ExpanderState = {
     openState:
       this.props.defaultOpen !== undefined ? this.props.defaultOpen : false,
   };
 
-  handleClick = () => {
-    const {
-      open,
-      onClick,
-      consumer: { onClick: consumerOnClick } = { onClick: undefined },
-      index,
-      expanderGroup,
-    } = this.props;
-    if (!!expanderGroup && !!consumerOnClick && index !== undefined) {
-      consumerOnClick(index);
-    } else {
-      const openState = !this.state.openState;
-      const notControlled = open === undefined;
-      if (notControlled) {
-        this.setState({ openState });
+  componentDidUpdate(prevProps: ExpanderProps, prevState: ExpanderState) {
```

Comment:
😅 This looks super complex conditional logic, but I guess there is no way around it. Is it possible that both blocks get executed on one update so that onClick and consumerOnClick are both executed? I was trying to figure out if an infinite update loop is possible in some condition, e.g. controlled open for one expander does not allow updating one item in group?

Reply:
- It is complex and maybe could be simplified. But at the moment I have no thought what to change. Maybe move the logics to separate functions?

- On the controlled example both are executed at once; onClick and consumerOnClick (now renamed to _onExpanderOpenChange_), so it should work okay™

Reply:
Sounds reasonable. And it is true that simplifying the logic might not be possible or might make readability even worse.